colorbleed/plugins/maya/load/load_yeti_cache.py

```diff
@@ -152,70 +219,56 @@ def validate_cache(self, filename, pattern="%04d"):
+            settings(list): list of dictionaries
 
-            # Create transform node
-            transform = node_settings["transform"]
-            transform_name = "{}:{}".format(namespace, transform["name"])
-            transform_node = cmds.createNode("transform", name=transform_name)
+        Returns:
+             list
 
-            lib.set_id(transform_node, transform["cbId"])
+        """
+
+        nodes = []
+        for node_settings in settings:
 
             # Create pgYetiMaya node
             original_node = node_settings["name"]
             node_name = "{}:{}".format(namespace, original_node)
-            yeti_node = cmds.createNode("pgYetiMaya",
-                                        name=node_name,
-                                        parent=transform_node)
+            yeti_node = cmds.createNode("pgYetiMaya", name=node_name)
 
-            lib.set_id(yeti_node, node_settings["cbId"])
-
-            nodes.append(transform_node)
-            nodes.append(yeti_node)
-
-            # Apply attributes to pgYetiMaya node
-            kwargs = {}
-            for attr, value in node_settings["attrs"].items():
-                if value is None:
-                    continue
+            # Create transform node
+            transform_node = node_name.rstrip("Shape")
 
-                attribute = "%s.%s" % (yeti_node, attr)
-                if isinstance(value, (str, unicode)):
-                    cmds.setAttr(attribute, value, type="string")
-                    continue
+            lib.set_id(transform_node, node_settings["transform"]["cbId"])
+            lib.set_id(yeti_node, node_settings["cbId"])
 
-                cmds.setAttr(attribute, value, **kwargs)
+            nodes.extend([transform_node, yeti_node])
 
             # Ensure the node has no namespace identifiers
-            node_name = original_node.replace(":", "_")
-
-            # Create full cache path
-            cache = os.path.join(self.fname, "{}.%04d.fur".format(node_name))
-            cache = os.path.normpath(cache)
-            cache_fname = self.validate_cache(cache)
-            cache_path = os.path.join(self.fname, cache_fname)
-
-            # Preset the viewport density
-            cmds.setAttr("%s.viewportDensity" % yeti_node, 0.1)
+            attributes = node_settings["attrs"]
+
+            # Check if cache file name is stored
+            if "cacheFileName" not in attributes:
+                file_name = original_node.replace(":", "_")
+                cache = os.path.join(self.fname,
+                                     "{}.%04d.fur".format(file_name))
+                self.validate_cache(cache)
+                attributes["cacheFileName"] = cache
+
+            # Update attributes with requirements
+            attributes.update({"viewportDensity": 0.1,
+                               "verbosity": 2,
+                               "fileMode": 1,
+                               "visibleInReflections": True,
```

Comment:
Nice to see this fixed - but I think it is actually more readable to keep these two "fixes" separate from the attributes update.

How about we do it explicitly like how Sven did it in his PR.

```python
            # Fix for : YETI-6
            # Fixes the render stats (this is literally taken from Perigrene's
            # ../scripts/pgYetiNode.mel script)
            cmds.setAttr(yeti_node + ".visibleInReflections", 1)
            cmds.setAttr(yeti_node + ".visibleInRefractions", 1)
```